behaviour/graql/reasoner/resolution/type-hierarchy.feature

```diff
@@ -0,0 +1,471 @@
+#
+# Copyright (C) 2020 Grakn Labs
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+Feature: Type Hierarchy Resolution
+
+  Background: Set up keyspaces for resolution testing
+
+    Given connection has been opened
+    Given connection delete all keyspaces
+    Given connection open sessions for keyspaces:
+      | materialised |
+      | reasoned     |
+    Given materialised keyspace is named: materialised
+    Given reasoned keyspace is named: reasoned
+
+
+  Scenario: when matching different roles to those that are actually inferred, no answers are returned
+    Given for each session, graql define
+      """
+      define
+
+      entity1 sub entity,
+          plays role1,
+          plays role2,
+          plays role3;
+
+      relation1 sub relation,
```

Comment:
this won't be legal anymore - can we rewrite schema to not share roles?

Reply:
Rewritten to test two cases:
`# Matching a sibling of the actual role`
and
`# Matching two siblings when only one is present`